Common/net/mbedtls_transport.c

```diff
@@ -407,7 +407,7 @@ static int mbedtls_ssl_send( void * pvCtx,
         }
     }
 
-    return ( int ) uxBytesSent;
+    return ( int ) lError < 0 ? lError : uxBytesSent;
```

Comment:
@ActoryOu Thanks for the fix. 

Is there a test case where this will get stuck due to the loop on line 361 or will there always be an error condition?